<launch>
 <!-- Robot description and state publishers / needed for EKF -->
 <param name="robot_description" command="cat '$(find mowgli)/description/mowgli.xml'" />
 <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" />
 <node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher" />

 <!-- external IMU filtering for EKF -->
 <node pkg="imu_filter_madgwick" type="imu_filter_node" name="imu_filter_madgwick_ekf" output="screen" respawn="false" >
        <param name="publish_tf" value="false"/>
        <param name="mag_bias_x" value="0.0"/>
        <param name="mag_bias_y" value="0.0" />
        <param name="mag_bias_z" value="0.0" />
        <param name="orientation_stddev" value="0.0"/>
        <param name="zeta" value="0.0" />
        <param name="gain" value="0.2" />
        <param name="use_mag" value="true" />
        <param name="use_magnetic_field_msg" value="true" />
        <remap from="/imu/data" to="/imu/data_dr"/> 
 </node>

 <!-- EKF -->
 <rosparam command="load" file="$(find mowgli)/params/ekf.yaml" />
 <node pkg="robot_localization" type="ekf_localization_node" name="ekf_se_odom" clear_params="true" />

</launch>
